.github/workflows/tests.yaml

```diff
@@ -0,0 +1,27 @@
+name: Tests
+
+on:
+  - push
+  - pull_request
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12']
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
```

Comment:
You should upgrade to the latest version of these actions. These versions rely on some outdated dependencies that GitHub is likely going to be removing from their runners soon.

Reply:
Gotcha. Should have used a more up to date tutorial :sob:

Reply:
Just to clarify, do you mean I should use v4 instead of v2?

Reply:
Okay things should be good now. Now i just need to get the tests passing lol

Reply:
> Just to clarify, do you mean I should use v4 instead of v2?

The latest version of `setup-python` is actually v5. v4 uses a newer but still EOL version of Node.

Reply:
Oh okay. I thought I saw v4 in the docs.